src/docfx/build/context/Cache.cs

```diff
@@ -41,6 +44,9 @@ internal class Cache
                 new Lazy<(List<Error>, TableOfContentsModel, List<Document>, List<Document>)>(
                     () => TableOfContentsParser.Load(context, file))).Value;
 
+        public Document GetDependentDocument(Docset docset, string pathToDocset, Func<Document> getDocumentFromGit)
```

Comment:
Since `getDocumentFromGit` is expensive, otherwise we wouldn't cache here. Is it possible multiple threads trying to create the same key? Or to say, do we need `ConcurrentDictionary<(Docset docset, string pathToDocset), Lazy<Document>>`?

Reply:
Good suggestion.